src/components/Tooltip/index.js

```diff
@@ -28,16 +32,23 @@ function Tooltip({
       onBlur: () => setIsActive(false),
     };
 
-  function handleDocumentClick(e) {
-    if (tooltipRef.current?.contains?.(e.target)) return;
+  const handleDocumentClick = useCallback((e) => {
+    if ([...additionalClickParents, tooltipRef.current].some((p) => p?.contains?.(e.target))) {
+      return;
+    }
     setIsActive(false);
-  }
+  }, [additionalClickParents]);
 
   useEffect(() => {
     document.addEventListener('click', handleDocumentClick, true);
     return () => document.removeEventListener('click', handleDocumentClick, true);
-  }, []);
+  }, [handleDocumentClick]);
+
+  useEffect(() => {
+    setIsActive(isOpened);
+  }, [isOpened]);
 
+  const classes = ['tyk-tooltip', className].filter(Boolean).join(' ');
```

Comment:
Trying to understand what `filter(Boolean)` does. Is it filtering out the falsy values , ie an empty string ?
If so , why would we have an empty string there, or e.g `null` ?

Reply:
Yep, it filters out the falsy values. It is useful for the case when you don't provide a `className`.
And this pattern is also used a lot when classes depend on a prop or some other condition because the `&&` expressions either short-circuit with a falsy value or they evaluate to the class name.
For example:
```js
const classes = [
  'default-class',
  isOpened && 'is-opened',
  age > 18 && 'is-adult',
].filter(Boolean).join(' ');
```